### PR TITLE
APOLLO-20773: Add feed-connector demo

### PR DESCRIPTION
**Pull request by mcondo**
- Added a basic feed-connector
- Please, see [README](https://github.com/lucidworks/connectors-sdk-resources/blob/2a0c70c0aaae73cb8548009ab7392a7f8caccb31/java-sdk/connectors/feed-connector/README.md) file
- Additionally, cleanup some readme files

### CODE REVIEW

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/config/FeedConfig.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  interface Properties extends ConnectorPluginProperties {
+
+    @Property(
+        title = "Feed file path",
+        description = "Feed file path location",
+        required = true,
+        order = 1
+    )
+    @StringSchema
+    String path();
```
Comment (acesar) [inline java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/config/FeedConfig.java]: A local file doesn't work easily in a cluster  - could you change it to point to blobStore item or maybe to be able to it read from gist 'https://gist.github.com/mcondo/1186e69d8bbf2779c4db30d43a52850f'.

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            .put("lastJobRunDateTime", lastJobRunDateTime)
+            .put("requestInfoId", fetchContext.getRequestInfo().getId())
+            .build()
+        )
+        .emit();
+  }
+
+  @Override
+  public PostFetchResult postFetch(PostFetchContext postFetchContext) {
+    return postFetchContext.newResult().withPurgeStrayItems();
```
Comment (acesar) [inline java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java]: Could you please code comments here? trying to explain what this part does?

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              .build()
+          )
+          .withTransient(true)
+          .emit();
+      logger.info("Emit candidate {} to fetch", id);
+    });
+  }
+
+  private void emitCheckpoint(FetchContext fetchContext, String id, long lastJobRunDateTime) {
+    logger.info("Emit checkpoint with date {}", lastJobRunDateTime);
```
Comment (acesar) [inline java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java]: Could you please code comments here?  why do we need checkpoint for the `PurgeStrayItems` to work?

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    Map<String, FeedEntry> entryMap = feed.getEntries();
+    entryMap.forEach((id, entry) -> {
+      fetchContext.newCandidate(entry.getId())
+          .withMetadata(ImmutableMap.<String, Object>builder()
+              .put("lastUpdatedEntry", entry.getLastUpdated())
+              .put("title", entry.getTitle())
+              .put("lastJobRunDateTime", lastJobRunDateTime)
+              .build()
+          )
+          .withTransient(true)
```
Comment (acesar) [inline java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java]: Could you please code comments here? why do we need transient candidate for the PurgeStrayItems to work?

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      emitCandidates(fetchContext, lastRunDateTime);
+      emitCheckpoint(fetchContext, connectorConfig.properties().path(), Instant.now().toEpochMilli());
+    } else {
+      processFeedEntry(fetchContext, input, metaData);
+    }
+    return fetchContext.newResult();
+  }
+
+  private void processFeedEntry(FetchContext fetchContext, FetchInput input, Map<String, Object> metaData) {
+    long lastRunDateTime = (Long) metaData.get("lastJobRunDateTime");
```
Comment (acesar) [inline java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/fetcher/FeedFetcher.java]: Could you please code comments here? - how/why are the `lastRunDateTime/lastJobRunDateTime` used/needed?

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/config/FeedConfig.java (hunk 1 above)
Comment (kaitillman-lucidworks) [reply java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/config/FeedConfig.java]: @mcondo What were your thoughts on Cesar's recommendation?

**Comment by roblucar** [comment]
Looks good. What is minimum Fusion version needed to build and run ?

File: java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/config/FeedConfig.java (hunk 1 above)
Comment (mcondo) [reply java-sdk/connectors/feed-connector/src/main/java/com/lucidworks/fusion/connector/plugin/config/FeedConfig.java]: @kaitillman-lucidworks 
I made changes to have two options:
- Use `path` to locate a json file.
- Use some `generate properties`, so the connector will generate entries. You can see the details in the readme file I added -> https://github.com/lucidworks/connectors-sdk-resources/pull/49/files#diff-5df815cd1be027a310f519d96066e29eR8
and
https://github.com/lucidworks/connectors-sdk-resources/pull/49/files#diff-5df815cd1be027a310f519d96066e29eR26

**Comment by mcondo** [comment]
@roblucar 
> Looks good. What is minimum Fusion version needed to build and run ?
Fusion 4.2.4

**Pull request merged**